### PR TITLE
Upgrade xz to 5.4.1

### PR DESCRIPTION
**Pull request by NobodyXu**
Signed-off-by: Jiahao XU <Jiahao_XU@outlook.com>

### CODE REVIEW

File: tests/xz.rs [hunk 1]
```diff
@@ -30,6 +30,14 @@ fn standard_files() {
...
             test_bad(&contents);
         } else {
+            // Ignores `good-1-arm64-lzma2-1.xz` and `good-1-arm64-lzma2-2.xz`
+            // since they uses the ARM64 filter.
+            //
+            // Ignores `good-1-empty-bcj-lzma2.xz` since it has an empty Block
+            // that uses PowerPC BCJ and LZMA2.
+            if filename.contains("arm64") || filename == "good-1-empty-bcj-lzma2.xz" {
+                continue;
+            }
```
Comment (NobodyXu) [inline tests/xz.rs:40]: I have no idea why these tests failed on my M1.

Disabling them fixed the CI, but not sure this is the right thing to do.

Comment (NobodyXu) [reply tests/xz.rs:40]: Aha I see this is because arm encoder is not enabled, which will be fixed by #111

**Comment by dragonmaus** [comment]
From what I can tell, the malicious actor started contributing to XZ [back in 2020](https://git.tukaani.org/?p=xz.git;a=commit;h=6468f7e41a8e9c611e4ba8d34e2175c5dacdbeb4).

[5.2.5](https://git.tukaani.org/?p=xz.git;a=blob;f=NEWS;h=2b75e5bebb3e2df6b9458ae92f7a0eedc58c670d;hb=2523c30705f49eabd27b854aa656ae87cc224808) looks like the last stable version before that point, so it might be worth holding off on any updates until some clarity is brought to the whole situation.

**Comment by NobodyXu** [comment]
Agreed, closing the PR for now.

**Pull request closed without merging**